api/observation.go

```diff
@@ -75,18 +77,18 @@ func (api *API) getObservations(w http.ResponseWriter, r *http.Request) {
 		}
 
 		// Get Version from dataset API
-		versionDoc, err := api.datasetClient.GetVersion(ctx, "", api.cfg.ServiceAuthToken, "", "", datasetID, edition, version)
+		versionDoc, err := api.datasetClient.GetVersion(ctx, userAuthToken, api.cfg.ServiceAuthToken, "", "", datasetID, edition, version)
 		if err != nil {
-			log.Event(ctx, "get observations: dataset api get /datasets/{dataset_id}/editions/{edition}/versions/{version} returned an error", log.ERROR, log.Error(err), logData)
+			log.Event(ctx, "get observations: dataset api failed to retrieve dataset version", log.ERROR, log.Error(err), logData)
 			return nil, err
 		}
 
 		// If not authorised, only published versions of datasets are accessible
 		if !authorised {
 			if versionDoc.State != dataset.StatePublished.String() {
-				logData["dataset_doc"] = datasetDoc
-				log.Event(ctx, "get observations: found no published dataset", log.ERROR, log.Error(errs.ErrDatasetNotFound), logData)
-				return nil, errs.ErrDatasetNotFound
+				logData["version_doc"] = datasetDoc
```

Comment:
im not sure this is right? should we be adding the versionDoc to `logData["version_doc"]`?

Reply:
good catch!